ui/nuxeo-document-comments/nuxeo-document-comment-thread.js

```diff
@@ -0,0 +1,311 @@
+/**
+@license
+(C) Copyright Nuxeo Corp. (http://nuxeo.com/)
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import '@polymer/iron-flex-layout/iron-flex-layout.js';
+import '@polymer/paper-icon-button/paper-icon-button.js';
+import '@polymer/paper-item/paper-icon-item.js';
+import '@polymer/paper-listbox/paper-listbox.js';
+import '@polymer/paper-menu-button/paper-menu-button.js';
+import '@nuxeo/nuxeo-elements/nuxeo-connection.js';
+import '@nuxeo/nuxeo-elements/nuxeo-element.js';
+import '@nuxeo/nuxeo-elements/nuxeo-resource.js';
+import '../widgets/nuxeo-tooltip.js';
+import '../widgets/nuxeo-user-avatar.js';
+// eslint-disable-next-line import/no-cycle
+import './nuxeo-document-comment.js';
+import './nuxeo-document-comments-styles.js';
+import { mixinBehaviors } from '@polymer/polymer/lib/legacy/class';
+import { html } from '@polymer/polymer/lib/utils/html-tag.js';
+import { FormatBehavior } from '../nuxeo-format-behavior.js';
+import CommentsMixin from './nuxeo-document-comments-mixin.js';
+
+/**
+ * Element to represent a thread of comments for a specific document (thread's root) on Nuxeo.
+ * It can represent multiple levels of comments and replies.
+ *
+ * Example:
+ *
+ *     <nuxeo-document-comment-thread uid="[[document.uid]]"></nuxeo-document-comment-thread>
+ *
+ * @appliesMixin Nuxeo.CommentsMixin
+ * @appliesMixin Nuxeo.FormatBehavior
+ * @memberof Nuxeo
+ * @demo https://nuxeo.github.io/nuxeo-elements/?path=/story/ui-nuxeo-document-comments--nuxeo-document-comment-thread
+ */
+{
+  class DocumentCommentThread extends CommentsMixin(mixinBehaviors([FormatBehavior], Nuxeo.Element)) {
+    static get template() {
+      return html`
+        <style include="nuxeo-document-comments-styles"></style>
+
+        <nuxeo-resource id="commentRequest" path="/id/[[uid]]/@comment/"></nuxeo-resource>
+
+        <template is="dom-if" if="[[_moreAvailable(comments.length, total, allCommentsLoaded)]]">
+          <span class="more-content no-selection pointer smaller" on-tap="_loadMore"
+            >[[_computeTextLabel(level, 'loadAll', total, i18n)]]</span
+          >
+        </template>
+        <template id="commentList" is="dom-repeat" items="[[comments]]" as="comment">
```

Comment:
Use `<dom-repeat...` instead, as we started using `<template is="dom-repeat` when we migrated to Polymer 2 (we kept it on Web UI, which was still in hybrid mode).

(add this to our checklist)